mne/time_frequency/tfr.py

```diff
@@ -1831,8 +2164,88 @@ def read_tfrs(fname, condition=None):
                              'The file contains "{1}""'
                              .format(condition, " or ".join(keys)))
         out = AverageTFR(**tfr_dict[condition])
-    elif is_average:
-        out = [AverageTFR(**d) for d in list(zip(*tfr_data))[1]]
     else:
-        out = [EpochsTFR(**d) for d in list(zip(*tfr_data))[1]]
+        inst = AverageTFR if is_average else EpochsTFR
+        out = [inst(**d) for d in list(zip(*tfr_data))[1]]
     return out
+
+
+def _is_numeric(n):
+    return isinstance(n, (int, float))
```

Comment:
Maybe better as `n = np.asarray(n); n.dtype.kind not in 'OSU' and n.shape == ()`

Reply:
... what

Reply:
What you wrote will fail for various `numpy` types. You can adapt what I wrote to work for them (maybe you also want to exclude complex in addition to object, string, and unicode (OSU))

Reply:
Hm. I think out of these:
https://docs.scipy.org/doc/numpy/reference/arrays.scalars.html
we want all the ints and floats, but not bool, complex, anything else. Wouldn't it be more economic to make an inclusive check than an exclusive check?

Reply:
I.e., `isinstance(n, (np.integer, np.floating))`

Reply:
Seems this was needed: `isinstance(n, (np.integer, np.floating, int, float))`